analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -79,86 +62,85 @@ protected sealed override SymbolReferenceInfo CreateMessage(SyntaxTree syntaxTre
             base.ShouldGenerateMetrics(tree)
             && !HasTooManyTokens(tree);
 
-        protected virtual SyntaxToken? GetSetKeyword(ISymbol valuePropertySymbol) => null;
-
-        protected static bool IsValuePropertyParameter(ISymbol symbol) =>
-            symbol is IParameterSymbol {IsImplicitlyDeclared: true, Name: "value"};
-
-        private SymbolReferenceInfo.Types.SymbolReference GetSymbolReference(SymRefInfo[] allReference, SyntaxTree tree)
+        private IEnumerable<ReferenceInfo> GetReferences(SyntaxNode root, SemanticModel model)
         {
-            TextSpan declarationSpan;
-            if (allReference.FirstOrDefault(r => r.IsDeclaration) is { } declaration)
-            {
-                declarationSpan = declaration.IdentifierToken.Span;
-            }
-            else
+            var references = new List<ReferenceInfo>();
+            var knownIdentifiers = new HashSet<string>();
+
+            foreach (var declaration in GetDeclarations(root))
             {
-                if (allReference.FirstOrDefault() is { } reference && GetSetKeyword(reference.Symbol) is { } setKeyword)
+                var declarationReferences = CreateDeclarationReferenceInfo(declaration, model);
+                if (declarationReferences == null)
                 {
-                    declarationSpan = setKeyword.Span;
+                    continue;
                 }
-                else
+
+                foreach (var reference in declarationReferences.Where(x => references.All(y => x.Node != y.Node)))
```

Comment:
`.Where(x => references.All(y => x.Node != y.Node)` is an iteration nested inside another. Maybe we could try to get rid of that logic. Either by introducing a `HashSet` of what is already processed and checking there or converting the references `List` to a `HashSet` and not have duplicates anymore.